tests/unit/s2n_server_key_share_extension_test.c

```diff
@@ -0,0 +1,318 @@
+    /* Test s2n_extensions_server_key_share_send */
+    {
+        struct s2n_connection *conn;
+
+        EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
+        struct s2n_stuffer* extension_stuffer = &conn->handshake.io;
+
+        /* Error if no curve have been selected */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_extensions_server_key_share_send(conn, extension_stuffer), S2N_ERR_NULL);
+
+        uint32_t cursor;
+        RECORD_WRITE_CURSOR(extension_stuffer, cursor);
+
+        for (int i = 0; i < S2N_ECC_SUPPORTED_CURVES_COUNT; i++) {
+            conn->secure.server_ecc_params.negotiated_curve = &s2n_ecc_supported_curves[i];
+            EXPECT_SUCCESS(s2n_extensions_server_key_share_send(conn, extension_stuffer));
+            EXPECT_STUFFER_BYTES_WRITTEN(extension_stuffer, s2n_ecc_supported_curves[i].share_size + 8, cursor);
+
+            EXPECT_EQUAL(conn->secure.server_ecc_params.negotiated_curve, &s2n_ecc_supported_curves[i]);
+            EXPECT_SUCCESS(s2n_ecc_params_free(&conn->secure.server_ecc_params));
+        }
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test s2n_extensions_server_key_share_recv with supported curves */
+    {
+        for (int i = 0; i < S2N_ECC_SUPPORTED_CURVES_COUNT; i++) {
+            struct s2n_connection *server_send_conn;
+            struct s2n_connection *client_recv_conn;
+
+            EXPECT_NOT_NULL(server_send_conn = s2n_connection_new(S2N_SERVER));
+            EXPECT_NOT_NULL(client_recv_conn = s2n_connection_new(S2N_CLIENT));
+
+            struct s2n_stuffer* extension_stuffer = &server_send_conn->handshake.io;
+
+            server_send_conn->secure.server_ecc_params.negotiated_curve = &s2n_ecc_supported_curves[i];
+            EXPECT_SUCCESS(s2n_extensions_server_key_share_send(server_send_conn, extension_stuffer));
+
+            STUFFER_READ_U16_EXPECT_EQUAL(extension_stuffer, TLS_EXTENSION_KEY_SHARE);
+            STUFFER_READ_U16_EXPECT_EQUAL(extension_stuffer, s2n_extensions_server_key_share_send_size(server_send_conn) - 2);
+
+            client_recv_conn->secure.client_ecc_params[i].negotiated_curve = &s2n_ecc_supported_curves[i];
+            s2n_ecc_generate_ephemeral_key(&client_recv_conn->secure.client_ecc_params[i]);
+
+            /* Parse key share */
+            EXPECT_SUCCESS(s2n_extensions_server_key_share_recv(client_recv_conn, extension_stuffer));
+            EXPECT_EQUAL(s2n_stuffer_data_available(extension_stuffer), 0);
+
+            EXPECT_EQUAL(server_send_conn->secure.server_ecc_params.negotiated_curve->iana_id, client_recv_conn->secure.server_ecc_params.negotiated_curve->iana_id);
+            EXPECT_EQUAL(server_send_conn->secure.server_ecc_params.negotiated_curve, &s2n_ecc_supported_curves[i]);
+
+            EXPECT_SUCCESS(s2n_connection_free(server_send_conn));
+            EXPECT_SUCCESS(s2n_connection_free(client_recv_conn));
+        }
+    }
+
+    /* Test s2n_extensions_server_key_share_recv with various payloads */
```

Comment:
Do these tests belong in this test class? Since the previous test tests that we can read/write all supported curves, these tests are really moving into s2n_ecc territory and testing whether s2n is properly parsing key data.